ui/src/RecentFilesDrawer/RecentFilesDrawer.tsx

```diff
@@ -48,16 +54,33 @@ export default function RecentFilesDrawer({
   const { curFlowID } = useContext(WorkspaceContext);
   const [selectedTag, setSelectedTag] = useState<string>();
   const [showAllTags, setShowAllTags] = useState(false);
-  useEffect(() => {
-    const all = listWorkflows();
-    setRecentFlow(all);
-  }, []);
+  const sortTypeRef = useRef<ESortTypes>(window.localStorage.getItem(sortTypeLocalStorageKey) as ESortTypes ?? ESortTypes.RECENTLY_MODIFIED)
 
   const onClickTag = (name: string) => {
     setSelectedTag(name);
     setRecentFlow(listWorkflows().filter((n) => n.tags?.includes(name)));
   };
 
+  const loadLatestWorkflows = () => {
+    const all = listWorkflows();
+    setRecentFlow(sortFlows(all, sortTypeRef.current));
+  }
+
+  const onSort = (type: ESortTypes) => {
+    setRecentFlow(sortFlows(recentFlows, type));
```

Comment:
prefer to move the sortFlows() into listWorkflows(sortby: ESortTypes) if there is sortby passed in.